commit 67b37247d50ca6e5508f100e24ebd3669ef0d916
Author: Jan Holesovsky <kendy@konik.suse.cz>
Date:   Tue Jan 19 10:30:59 2010 +0100

    Introduce 'cz(dvorak-ucw)' layout.
    
    It is a US Dvorak layout with the CZ accented characters accessible using
    AltGr under the appropriate keys (eg. AltGr+a => aacute, etc.).

diff --git a/rules/base.xml.in b/rules/base.xml.in
index 4db5667..c457d68 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -2059,6 +2059,12 @@
 	    <_description>UCW layout (accented letters only)</_description>
 	 </configItem>
 	</variant>
+	<variant>
+	  <configItem>
+	    <name>dvorak-ucw</name>
+	    <_description>US Dvorak with CZ UCW support</_description>
+	 </configItem>
+	</variant>
       </variantList>
     </layout>
     <layout>
diff --git a/symbols/cz b/symbols/cz
index cd2e09e..d761b75 100644
--- a/symbols/cz
+++ b/symbols/cz
@@ -142,3 +142,33 @@ xkb_symbols "ucw" {
     key <AD09> { [oacute,      Oacute      ] };
     key <AD10> { [odiaeresis,  Odiaeresis  ] };
 };
+
+partial alphanumeric_keys
+xkb_symbols "dvorak-ucw" {
+
+    // US Dvorak layout extend with UCW - AltGr produces accented characters
+
+    include "us(dvorak)"
+
+    name[Group1]= "Czechia - US Dvorak with CZ UCW support";
+
+    key <AB03> { [ j, J, eacute, Eacute ] };
+    key <AB04> { [ k, K, uacute, Uacute ] };
+    key <AB10> { [ z, Z, zcaron, Zcaron ] };
+
+    key <AC01> { [ a, A, aacute, Aacute ] };
+    key <AC02> { [ o, O, oacute, Oacute ] };
+    key <AC03> { [ e, E, ecaron, Ecaron ] };
+    key <AC04> { [ u, U, uring,  Uring  ] };
+    key <AC05> { [ i, I, iacute, Iacute ] };
+    key <AC06> { [ d, D, dcaron, Dcaron ] };
+    key <AC08> { [ t, T, tcaron, Tcaron ] };
+    key <AC09> { [ n, N, ncaron, Ncaron ] };
+    key <AC10> { [ s, S, scaron, Scaron ] };
+
+    key <AD05> { [ y, Y, yacute, Yacute ] };
+    key <AD08> { [ c, C, ccaron, Ccaron ] };
+    key <AD09> { [ r, R, rcaron, Rcaron ] };
+
+    include "level3(ralt_switch)"
+};
